--- a/cmd/nvidia-cdi-hook/cudacompat/cudacompat.go
+++ b/cmd/nvidia-cdi-hook/cudacompat/cudacompat.go
@@ -155,7 +155,7 @@
 
 	libs, err := containerRoot.globFiles(filepath.Join(o.cudaCompatContainerRoot, "libcuda.so.*.*"))
 	if err != nil {
-		m.logger.Warningf("Failed to find CUDA compat library: %w", err)
+		m.logger.Warningf("Failed to find CUDA compat library: %v", err)
 		return "", nil
 	}
 
@@ -216,7 +216,7 @@
 	// libraries in the container should be used over the host driver libraries.
 	cudaCompatHeader, err := GetCUDACompatElfHeaderFromReader(libcudaCompatFile)
 	if err != nil {
-		m.logger.Warningf("failed to get ELF header from CUDA compat library: %w", err)
+		m.logger.Warningf("failed to get ELF header from CUDA compat library: %v", err)
 	}
 	if cudaCompatHeader != nil {
 		return cudaCompatHeader.UseCompat(compatDriverSemver, hostDriverSemver, hostCUDASemver), nil
--- a/pkg/nvcdi/driver-wsl.go
+++ b/pkg/nvcdi/driver-wsl.go
@@ -53,7 +53,7 @@
 	}
 	defer func() {
 		if err := dxcore.Shutdown(); err != nil {
-			l.logger.Warningf("failed to shutdown dxcore: %w", err)
+			l.logger.Warningf("failed to shutdown dxcore: %v", err)
 		}
 	}()
 
